lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -1011,7 +1012,11 @@ static ParseResult parseStructExtractOp(OpAsmParser &parser,
       parser.parseColonType(declType))
     return failure();
 
-  Type resultType = declType.getFieldType(fieldName.getValue());
+  auto structType = getCanonicalType(declType).dyn_cast<StructType>();
+  if (!structType)
```

Comment:
The type constraint ensures this is a struct type.

Reply:
That's what I thought, but if you intentionally feed the parser not a struct type, this code needs to check before calling `getFieldType` below. The existing error test demonstrates this.